packages/nbdime/src/common/mergeview.ts

```diff
@@ -143,563 +341,404 @@ function createNbdimeMergeView(
+        if (!decorationKey.includes('Merge')) {
+          effects = effects.concat(this.createGutterEffects(editor, chunk, chunkLastLine, true, 'picker'));
+
         } else if (conflict) {
-          // Add conflict markers on editor, if conflicted
-          let conflictMarker = elt('div', CONFLICT_MARKER, '');
-          (conflictMarker as any).sources = sources;
-          conflictMarker.classList.add(GUTTER_CONFLICT_CLASS);
-          editor.setGutterMarker(line, GUTTER_CONFLICT_CLASS, conflictMarker);
+          effects = effects.concat(this.createGutterEffects(editor, chunk, startingOffset, true, 'picker'));
         }
-        editor.addLineClass(line, 'background', classes.end + '-empty');
-        markers.push(line);
       }
     }
-    let cls = type === DIFF_OP.DIFF_DELETE ? classes.del : classes.insert;
-    editor.operation(function() {
-      let edit = editor === self.baseEditor;
-      if (self.classes) {
-        clearMarks(editor, markers, classes);
-      } else {
-        clearMergeMarks(editor, markers);
-      }
-      highlightChars(editor, diff, markers, cls);
-      for (let c of self.chunks) {
-        if (edit) {
-          markChunk(editor, c.baseFrom, c.baseTo, c.sources);
-        } else {
-          markChunk(editor, c.remoteFrom, c.remoteTo, c.sources);
-        }
-      }
-    });
+    return effects;
   }
 
-  get ownEditor(): CodeMirror.Editor {
-    return this.ownWidget.cm;
-  }
+  private clearLineEffects(editor: EditorView, chunkArray: Chunk[]) {
+    let effects: StateEffect<unknown>[] = [];
 
-  ownWidget: EditorWidget;
-  model: IStringDiffModel;
-  type: string;
-  showDifferences: boolean;
-  dealigned: boolean;
-  forceUpdate: Function;
-  baseEditor: CodeMirror.Editor;
-  chunks: Chunk[];
-  lineChunks: Chunk[];
-  gap: HTMLElement;
-  lockScroll: boolean;
-  updating: boolean;
-  updatingFast: boolean;
-  collapsedRanges: {line: number, size: number}[] = [];
-
-  protected updateCallback: (force?: boolean) => void;
-  protected copyButtons: HTMLElement;
-  protected lockButton: HTMLElement;
-  protected classes: DiffClasses | null;
-}
-
-
-// Updating the marks for editor content
-
-function clearMergeMarks(editor: CodeMirror.Editor, arr: Marker[]) {
-  for (let postfix of ['-local', '-remote', '-either', '-custom']) {
-    let classes = copyObj(mergeClassPrefix);
-    for (let k of Object.keys(classes)) {
-      classes[k] += postfix;
+    for (let chunk of chunkArray) {
+      let sources: ChunkSource[] = chunk.sources;
+      let decorationKey = this.getDecorationKey(sources);
+      effects.push(removeHighlightEffect.of({highlightType: 'chunk', decorationKey: decorationKey}));
+      effects.push(removeHighlightEffect.of({highlightType: 'conflict', decorationKey: decorationKey}));
+      effects.push(removeHighlightEffect.of({highlightType: 'start', decorationKey: decorationKey}));
+      effects.push(removeHighlightEffect.of({highlightType: 'end', decorationKey: decorationKey}));
```

Comment:
This can be shorten with a `for ... of` loop

Reply:
Thanks. This has been changed.